src/function/mtk.jl

```diff
@@ -12,8 +17,8 @@ function instantiate_function(f, x, ::AutoModelingToolkit, p, num_cons=0)
     end
 
     if f.hess === nothing
-        hess_oop, hess_iip = ModelingToolkit.generate_hessian(sys, expression=Val{false})
-        hess(J, u) = (hess_iip(J, u, p); J)
+        hess_oop, hess_iip = ModelingToolkit.generate_hessian(sys, expression=Val{false}, sparse = ad.obj_sparse)
```

Comment:
Unless we set `hes_prototype` somewhere, this is going to be incorrect. Can you add prototype fields in https://github.com/SciML/SciMLBase.jl/blob/master/src/problems/basic_problems.jl#L220 ? Then we'll need to pass these onto the solvers.